jmbitcoin/jmbitcoin/secp256k1_transaction.py

```diff
@@ -837,48 +193,36 @@ def mktx(ins, outs, version=1, locktime=0):
     else:
         sequence = 0xffffffff
     for i in ins:
-        if isinstance(i, dict) and "outpoint" in i:
-            txobj["ins"].append(i)
-        else:
-            if isinstance(i, dict) and "output" in i:
-                i = i["output"]
-            txobj["ins"].append({
-                "outpoint": {"hash": i[:64],
-                             "index": int(i[65:])},
-                "script": "",
-                "sequence": sequence
-            })
+        outpoint = CMutableOutPoint((i[0][::-1]), i[1])
+        inp = CMutableTxIn(prevout=outpoint, nSequence=sequence)
+        vin.append(inp)
     for o in outs:
-        if isinstance(o, str):
-            addr = o[:o.find(':')]
-            val = int(o[o.find(':') + 1:])
-            o = {}
-            if re.match('^[0-9a-fA-F]*$', addr):
-                o["script"] = addr
-            else:
-                o["address"] = addr
-            o["value"] = val
-
-        outobj = {}
-        if "address" in o:
-            outobj["script"] = address_to_script(o["address"])
-        elif "script" in o:
-            outobj["script"] = o["script"]
-        else:
-            raise Exception("Could not find 'address' or 'script' in output.")
-        outobj["value"] = o["value"]
-        txobj["outs"].append(outobj)
-    return serialize(txobj)
-
-def make_shuffled_tx(ins, outs, deser=True, version=1, locktime=0):
-    """ Simple utility to ensure transaction
+        # note the to_scriptPubKey method is only available for standard
+        # address types
+        out = CMutableTxOut(o["value"],
+                    CCoinAddress(o["address"]).to_scriptPubKey())
+        vout.append(out)
+    return CMutableTransaction(vin, vout, nLockTime=locktime)
```

Comment:
Note that there's slightly more convenient, but a bit more wasteful way to create mutable transaction by constructing with non-mutable parts, and doing `return tx.to_mutable()` afterwards. `tx.to_immutable()` also exists. These methods return new instances (unless the expected conversion is immutable -> immutable).

Reply:
Noted. I have tried to just work with CMutableTransaction here, but perhaps more thought should be put into situations where the immutable version is desirable. Certainly performance isn't an issue (the only place that matters in in wallet scanning, so addresses from bip32 master secret creation, but our using libsecp mostly handles that).